gammapy/spectrum/models.py

```diff
@@ -574,30 +577,131 @@ class TableModel(SpectralModel):
         Array with the values of the model at energies ``energy``.
     amplitude : float
         Model amplitude that is multiplied to the supplied arrays. Defaults to 1.
+    scale_logy : boolean
+        interpolation can be done linearly or in logarithm
     """
 
-    def __init__(self, energy, values, amplitude=1):
+    def __init__(self, energy, values, amplitude=1, scale_logy=True):
         from scipy.interpolate import interp1d
         self.parameters = Bunch(amplitude=amplitude)
         self.energy = energy
         self.values = values
+        self.scale_logy = scale_logy
 
         loge = np.log10(self.energy.to('eV').value)
         try:
             self.unit = self.values.unit
-            logy = np.log10(self.values.value)
+            if scale_logy is True:
+                y = np.log10(self.values.value)
+            else:
+                y = self.values.value
         except AttributeError:
             self.unit = u.Unit('')
-            logy = np.log10(self.values)
+            if scale_logy is True:
+                y = np.log10(self.values)
+            else:
+                y = self.values
+        self.interpy = interp1d(loge,
+                                y,
+                                fill_value=-np.Inf,
+                                bounds_error=False,
+                                kind='cubic')
 
-        self.interplogy = interp1d(loge,
-                                   logy,
-                                   fill_value=-np.Inf,
-                                   bounds_error=False,
-                                   kind='cubic')
+    @classmethod
+    def from_file(cls, filename, param):
+        """A Table containing aborbed values from a XSPEC model
+        as a function of energy.
+        Todo:
+        Format of the file should be described and discussed in
+        https://gamma-astro-data-formats.readthedocs.io/en/latest/index.html
+
+        Parameters
+        ----------
+        filename : `str`
+            File containing the XSPEC model
+        param : float
+            Model parameter value
+        Examples
```

Comment:
I don't think this docstring will render correctly.
You need to put an empty line before the `Examples` heading, and I think also colon before the code block.

Can you please check locally with
```
python setup.py build_docs
open docs/_build/html/index.html
```
and then check this docstring on the HGPS page?

We will add automated checks for docstrings in the future, there's a discussion for Astropy generally how to best do it. For now, checking docstrings is done manually.